services/rfq/relayer/quoter/quoter.go

```diff
@@ -519,17 +550,32 @@ func (m *Manager) getOriginAmount(parentCtx context.Context, origin, dest int, a
 		quoteAmount = minQuoteAmount
 	}
 
-	// Finally, clip the quoteAmount by the balance
-	if quoteAmount.Cmp(balance) > 0 {
-		span.AddEvent("quote amount greater than balance", trace.WithAttributes(
+	// Clip the quoteAmount by the max origin balance
+	maxBalance := m.config.GetMaxBalance(input.OriginChainID, input.OriginTokenAddr)
+	if input.OriginBalance != nil && maxBalance.Cmp(big.NewInt(0)) > 0 {
+		quotableBalance := new(big.Int).Sub(maxBalance, input.OriginBalance)
```

Comment:
I think we need to check if `quotableBalance < 0` here, as we could assign a negative value to `quoteAmount` below should the balance exceed maximum.